src/models/databaseModel.ts

```diff
@@ -297,272 +37,26 @@ export enum DbStatePathType {
-    isCancelTimelockExpiredDbState(dbState.state) &&
-    dbState.type === DbStateType.CANCEL_TIMELOCK_EXPIRED
-  );
+export function getSwapXmrAmount(swap: GetSwapInfoResponse): number {
+  return piconerosToXmr(swap.xmrAmount);
 }
 
-export interface MergedBtcCancelledDbState extends MergedDbState {
-  type: DbStateType.BTC_CANCELLED;
-  state: ExecutionSetupDoneDbState & BtcLockedDbState & BtcCancelledDbState;
-}
-
-export function isMergedBtcCancelledDbState(
-  dbState: MergedDbState
-): dbState is MergedBtcCancelledDbState {
-  return (
-    isExecutionSetupDoneDbState(dbState.state) &&
-    isBtcLockedDbState(dbState.state) &&
-    isBtcCancelledDbState(dbState.state) &&
-    dbState.type === DbStateType.BTC_CANCELLED
-  );
-}
-
-export interface MergedDoneBtcRefundedDbState extends MergedDbState {
-  type: DbStateType.DONE_BTC_REFUNDED;
-  state: ExecutionSetupDoneDbState & BtcLockedDbState & DoneBtcRefundedDbState;
-}
-
-export function isMergedDoneBtcRefundedDbState(
-  dbState: MergedDbState
-): dbState is MergedDoneBtcRefundedDbState {
-  return (
-    isExecutionSetupDoneDbState(dbState.state) &&
-    isBtcLockedDbState(dbState.state) &&
-    isDoneBtcRefundedDbState(dbState.state) &&
-    dbState.type === DbStateType.DONE_BTC_REFUNDED
-  );
-}
-
-export interface MergedDoneBtcPunishedDbState extends MergedDbState {
-  type: DbStateType.DONE_BTC_PUNISHED;
-  state: ExecutionSetupDoneDbState & BtcLockedDbState & DoneBtcPunishedDbState;
-}
-
-export function isMergedDoneBtcPunishedDbState(
-  dbState: MergedDbState
-): dbState is MergedDoneBtcPunishedDbState {
-  return (
-    isExecutionSetupDoneDbState(dbState.state) &&
-    isBtcLockedDbState(dbState.state) &&
-    isDoneBtcPunishedDbState(dbState.state) &&
-    dbState.type === DbStateType.DONE_BTC_PUNISHED
-  );
-}
-
-export function getSwapTxFees(dbState: MergedDbState): number {
-  const tx = dbState.state.Bob.ExecutionSetupDone.state2.tx_lock;
-
-  const sumInput = tx.inner.inputs
-    .map((input) => input.witness_utxo.value)
-    .reduce((prev, next) => prev + next);
-
-  const sumOutput = tx.inner.unsigned_tx.output
-    .map((output) => output.value)
-    .reduce((prev, next) => prev + next);
-
-  return satsToBtc(sumInput - sumOutput);
-}
-
-export function getSwapBtcAmount(dbState: MergedDbState): number {
-  return satsToBtc(
-    dbState.state.Bob.ExecutionSetupDone.state2.tx_lock.inner.unsigned_tx
-      .output[0]?.value
-  );
-}
-
-export function getSwapXmrAmount(dbState: MergedDbState): number {
-  return pionerosToXmr(dbState.state.Bob.ExecutionSetupDone.state2.xmr);
-}
-
-export function getSwapExchangeRate(dbState: MergedDbState): number {
-  const btcAmount = getSwapBtcAmount(dbState);
-  const xmrAmount = getSwapXmrAmount(dbState);
+export function getSwapExchangeRate(swap: GetSwapInfoResponse): number {
+  const btcAmount = getSwapBtcAmount(swap);
+  const xmrAmount = getSwapXmrAmount(swap);
 
   return btcAmount / xmrAmount;
 }
```

Comment:
**Suggestion:** The function `getSwapExchangeRate` doesn't handle the case where `xmrAmount` is 0, which would cause a division by zero error. It would be better to add a check for this case.
```suggestion
export function getSwapExchangeRate(swap: GetSwapInfoResponse): number {
  const btcAmount = getSwapBtcAmount(swap);
  const xmrAmount = getSwapXmrAmount(swap);

  if (xmrAmount === 0) {
    throw new Error('XMR amount is zero, cannot calculate exchange rate');
  }

  return btcAmount / xmrAmount;
}
```